p4include/v1model.p4

```diff
@@ -619,6 +619,14 @@ extern void assert(in bool check);
  */
 extern void assume(in bool check);
 
+/*
+ * Log user defined messages
+ * Example: log_msg("User defined message");
+ * or log_msg("Value1 = {}, Value2 = {}",[value1, value2]);
+ */
+extern void log_msg<T>(in string msg);
```

Comment:
We could also make the first argument directionless, to enforce the requirement for it to be a compile-time constant.